src/careamics_restoration/engine.py

```diff
@@ -201,48 +198,68 @@ def train(
             )
 
         eval_loader = self.get_val_dataloader(val_path)
-
         self.logger.info(f"Starting training for {self.cfg.training.num_epochs} epochs")
 
         val_losses = []
+
         try:
-            for epoch in self.progress(
-                range(self.cfg.training.num_epochs),
-                task_name="Epochs",
-                overall_progress=True,
-            ):  # loop over the dataset multiple times
-                train_outputs = self._train_single_epoch(
+            train_stats = []
+            eval_stats = []
+
+            # loop over the dataset multiple times
+            for epoch in range(self.cfg.training.num_epochs):
+                try:
+                    epoch_size = epoch_size
```

Comment:
`epoch_size` is defined nowhere

Reply:
Was this solved?

Reply:
That's why it's wrapper into try except. Logic is the 1st epoch progress bar has no length, starting from the 2nd it will have the length